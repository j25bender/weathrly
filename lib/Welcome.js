import React from 'react';
import App from '../lib/App.js';
import '../styles/Welcome.css';
import autoData from '../lib/AutoCompleteData.js';

export default class Welcome extends React.Component {
    constructor () {
        super();

        this.state = {
            inputValue: '',
            CurrentWeather: {},
            SevenHourForecast: [],
            TenDayForecast: []
        }
        this.handleInputChange = this.handleInputChange.bind(this);
        this.handleBtnClick = this.handleBtnClick.bind(this);
    }

    handleInputChange (e) {
        this.setState({
            inputValue: e.target.value
        })
    }

    handleBtnClick () {
        if(isNaN(this.state.inputValue) 
            && autoData.data.includes(this.state.inputValue) 
            || this.state.inputValue.length === 5 && !isNaN(this.state.inputValue)) {

            this.props.searchValue(this.state.inputValue);
            this.setState({inputValue: this.state.inputValue})
            localStorage.setItem("location", this.state.inputValue);
        } else { alert('Invalid Input: Please Enter City, State OR Zip Code e.g. Denver, CO OR 80202')}       
    }

    render() {
        return (
            <div className="welcomeContainer">
                <div className="welcome">
                    <h4 className="welcomeText">Welcome to Weathrly!</h4>
                    <input id="searchInput" 
                           value={this.state.inputValue}
                           type="text"
                           onChange={this.handleInputChange}
                           autoFocus 
                    />
                    <button onClick={this.handleBtnClick}>Search by City or Zip</button>
                </div>
            </div>
        )
    }
}