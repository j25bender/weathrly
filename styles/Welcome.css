.searchContainer {
    height: 10vh;
}

.welcome {
    text-align: center;
    background-color: lightcyan;
    width: 90vw;
    height: 10vh;
    margin: auto;
    top: 0;
}

.container {
    position: relative;
    top: 50%;
    transform: translateY(-50%);
}