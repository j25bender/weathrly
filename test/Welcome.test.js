import React from 'react';
import { shallow, mount } from 'enzyme';
import Welcome from "../lib/Welcome";
import cleanData from '../lib/cleanData';
import mockData from '../lib/mockData';
import 'jest-localstorage-mock';

describe('Welcome Tests ', () => {
    let filteredData;
    let wrapper;
    
    beforeEach(() => {
        filteredData = cleanData(mockData);
        wrapper = mount(<Welcome />);
    });

    it('Welcome component should exist', () => {
        expect(wrapper).toBeDefined()
        expect(wrapper.find('.welcomeText').text()).toEqual('Welcome to Weathrly!')        
    });

    it('Welcome should render 1 Welcome component', () => {
        expect(wrapper.find('Welcome').length).toEqual(1);
        expect(Welcome).toMatchSnapshot();
    });
  
    it('Should call handleBtnClick when button is clicked', () => {
        const submitButton = wrapper.find('button')
        const inputValue = wrapper.find('input')

        wrapper.instance().handleBtnClick = jest.fn()
        
        inputValue.simulate('change', { target: { value: 'Denver, CO'}})
        submitButton.simulate('click')
        
        expect(wrapper.instance().handleBtnClick).toHaveBeenCalledTimes(1)

        inputValue.simulate('change', { target: { value: 'Boston, MA'}})
        submitButton.simulate('click')
//state change test
        expect(wrapper.instance().handleBtnClick).toHaveBeenCalledTimes(2)
    });
});