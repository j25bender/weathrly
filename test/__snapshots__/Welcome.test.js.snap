// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Tests  Welcome should render 1 Welcome component 1`] = `[Function]`;
